addon/addon.go

```diff
@@ -149,20 +152,35 @@ func NewAddonManager(kubeConfig *rest.Config, agentImage string, agentInstallAll
 		return nil, err
 	}
 
+	addonClient, err := addonv1alpha1client.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
 	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
 	if err != nil {
 		klog.Errorf("unable to create kube client: %v", err)
 		return addonMgr, err
 	}
 
 	agentFactory := addonfactory.NewAgentAddonFactory(AppMgrAddonName, ChartFS, ChartDir).
-		WithGetValuesFuncs(getValue, addonfactory.GetValuesFromAddonAnnotation).
+		// register the supported configuration types
+		WithConfigGVRs(
+			schema.GroupVersionResource{Version: "v1", Resource: "configmaps"},
```

Comment:
Do we use `configmap` as the configuration for application addon? if we do not, we do not need to add the `configmaps`